### PR TITLE
 feat(react-i18n): add useTranslate hook

### PR DESCRIPTION
**Pull request by flepretre**
### useTranslate hook

This hook provide the `t` function in a functional component.

```jsx harmony
import React from 'react';
import { useTranslate } from '@m6web/react-i18n';

// Interpolation values
const data = { element: 'foo' };

export const MyComponent = ({ nbExample }) => {
  const t = useTranslate();

  return (
    <div class="foo">
      <h1>
        {t('foo.bar')}
      </h1>
      <p>
        {t('foo.exemple', data, nbExample, true)}
      </p>
    </div>
  );
};
```

### CODE REVIEW

File: packages/react-i18n/src/hooks/useT.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { useContext } from 'react';
+import { Context } from '../context/i18n.context';
+
+export const useT = () => useContext(Context);
```
Comment (fdubost) [inline packages/react-i18n/src/hooks/useT.js]: I prefer `useTranslate` 👼

File: packages/react-i18n/README.md [hunk 2]
```diff
@@ -137,12 +137,39 @@ export default translate(MyComponent);
 
 Note that **number** and **data** can be used together.
 
+### useTranslate hook
+
+This hook provide the `t` function in a functional component.
```
Comment (Exodu5) [inline packages/react-i18n/README.md]: `provides` ?

File: packages/react-i18n/src/hooks/useTranslate.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { useContext } from 'react';
+import { Context } from '../context/i18n.context';
+
+export const useTranslate = () => useContext(Context);
```
Comment (Slashgear) [inline packages/react-i18n/src/hooks/useTranslate.js]: Maybe we could `useMemo` to avoid multiple calls on translate function ?

Comment (flepretre) [reply packages/react-i18n/src/hooks/useTranslate.js]: 🤔 not sure, it looks like micro optimisation to me.
If it brinks perf issues then we will.

**Pull request merged**